mmaction/core/runner/omnisource_runner.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Open-MMLab. All rights reserved.
+import time
+import warnings
+
+import mmcv
+from mmcv.runner import EpochBasedRunner, Hook
+from mmcv.runner.utils import get_host_info
+
+
+def cycle(iterable):
+    iterator = iter(iterable)
+    while True:
+        try:
+            yield next(iterator)
+        except StopIteration:
+            iterator = iter(iterable)
+
+
```

Comment:
`@HOOKS.register_module()`

Reply:
This class doesn't need registration, since we directly use it without calling any build_xxx function.